src/oic/oic/provider.py

```diff
@@ -2049,3 +1883,449 @@ def get_by_sub_and_(self, sub: str, key: str, val: Any) -> Optional[str]:
+            "key": self.sdb.get_verify_logout(uid),
+            "redirect": redirect,
+            "action": "/" + EndSessionEndpoint("").etype,
+        }
+        return Response(
+            self.template_renderer("verify_logout", context), headers=headers
+        )
+
+    def _get_uid_from_cookie(
+        self, cookie: Union[str, SimpleCookie, None]
+    ) -> Tuple[Optional[CookieDealer], Optional[str], Optional[str]]:
+        """
+        Get cookie_dealer, client_id and uid from cookie.
+
+        :param cookie: Received cookie
+        :return: Tuple containing CookieDealer instance, client ID and User ID
+        """
+        if cookie is None:
+            return None, None, None
+
+        cookie_dealer = CookieDealer(srv=self)
+        client_id = uid = None
+
+        _cval = cookie_dealer.get_cookie_value(cookie, self.sso_cookie_name)
+        if _cval:
+            (value, _ts, typ) = _cval
+            if typ == "sso":
+                uid, client_id = value.split(DELIM)
+
+        return cookie_dealer, client_id, uid
+
+    def do_back_channel_logout(
+        self, cinfo: dict, sub: str, sid: str
+    ) -> Optional[Tuple[str, str]]:
+        """
+        Prepare information to be used to do a back-channel logout.
+
+        :param cinfo: Client information
+        :param sub: Subject identifier
+        :param sid: The Issuer ID
+        :return: Tuple with logout URI and signed logout token
+        """
+        try:
+            back_channel_logout_uri = cinfo["backchannel_logout_uri"]
+        except KeyError:
+            return None
+
+        # always include sub and sid so I don't check for
+        # backchannel_logout_session_required
+
+        payload = {
+            "sub": sub,
+            "sid": sid,
+            "events": {BACK_CHANNEL_LOGOUT_EVENT: {}},
+            "jti": uuid.uuid4().hex,
+        }
+
+        try:
+            alg = cinfo["id_token_signed_response_alg"]
+        except KeyError:
+            alg = self.capabilities["id_token_signing_alg_values_supported"][0]
+
+        _jws = JWT(self.keyjar, iss=self.name, lifetime=86400, sign_alg=alg)
+        sjwt = _jws.pack(aud=cinfo["client_id"], **payload)
+
+        return back_channel_logout_uri, sjwt
+
+    def clean_sessions(self, usids: List[str]):
+        """
+        Remove Session IDs from the session DB.
+
+        :param usids: List of session IDs
+        """
+        _sdb = self.sdb
+        # Clean out all sessions
+        for sid in usids:
+            del _sdb[sid]
+
+    def logout_all_clients(self, sid: str) -> Dict:
+        """
+        Collect information necessary to logout one user from all clients he/she has been using.
+
+        :param sid: Session ID
+        :return: Dictionary with logout information
+        """
+        _sdb = self.sdb
+
+        uid = _sdb.get_uid_by_sid(sid)
+        _client_sid = {}
+        # Find all the session IDs this user has gotten
+        usids = session_get(_sdb, "uid", uid)
```

Comment:
I think we do not need to use `sessoin_get` for provider, since using anything else than proper SDB is discouraged for two releases.

Reply:
You previously marked this as a breaking change and therefor something that should be avoided. Has that changed ?
It was the whole reason for me to do all session_* functions.

Reply:
OK